From 793eda7b258a562757b51c96044b103b638f1a63 Mon Sep 17 00:00:00 2001
From: Collin Funk <collin.funk1@gmail.com>
Date: Thu, 1 May 2025 22:20:58 -0700
Subject: [PATCH] tests: Fix link errors for t-thread-local.

On platforms where pthread_create is not in libc t-thread-local fails to
link.  Issue found on NetBSD 10.0.
* tests/Makefile.am (t_thread_local_LDADD): Add $(standard_ldadd),
$(GPG_ERROR_MT_LIBS), and @LDADD_FOR_TESTS_KLUDGE@.
(t_thread_local_CFLAGS): Add $(GPG_ERROR_MT_CFLAGS).

--

GnuPG-bug-id: 7634
Signed-off-by: Collin Funk <collin.funk1@gmail.com>
---
 tests/Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index 3170a58ea..cef9b7006 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -94,6 +94,8 @@ t_secmem_LDADD = $(standard_ldadd) @LDADD_FOR_TESTS_KLUDGE@
 testapi_LDADD = $(standard_ldadd) @LDADD_FOR_TESTS_KLUDGE@
 t_lock_LDADD = $(standard_ldadd) $(GPG_ERROR_MT_LIBS) @LDADD_FOR_TESTS_KLUDGE@
 t_lock_CFLAGS = $(GPG_ERROR_MT_CFLAGS)
+t_thread_local_LDADD = $(standard_ldadd) $(GPG_ERROR_MT_LIBS) @LDADD_FOR_TESTS_KLUDGE@
+t_thread_local_CFLAGS = $(GPG_ERROR_MT_CFLAGS)
 testdrv_LDADD = $(LDADD_FOR_TESTS_KLUDGE)
 
 # Build a version of the test driver for the build platform.
